File: javascript/tarai.js
let tarai = (x) => (y) => x <= y ? (_) => y:(z) => tarai(tarai(x - 1)(y)(z))(tarai(y - 1)(z)(x))(tarai(z - 1)(x)(y));
console.log(tarai(12)(6)(0));

File: lua/tarai.lua
local function tarai(x)
  return function(y)
    if x <= y then
      return function(_)
        return y;
      end;
    else
      return function(z)
        return tarai(tarai(x - 1)(y)(z))(tarai(y - 1)(z)(x))(tarai(z - 1)(x)(y));
      end;
    end
  end;
end

print (tarai(12)(6)(0))